clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -0,0 +1,202 @@
+			return false
+		}
+	}
+	return true
+}
+
+func (a *apiAvailabilityMeasurement) start(config *measurement.MeasurementConfig, SSHToMasterSupported bool, probeDuration int) ([]measurement.Summary, error) {
+	a.hosts = config.ClusterFramework.GetClusterConfig().MasterIPs
+	if len(a.hosts) < 1 {
+		return nil, goerrors.Errorf("Api Server measurement can't execute due to no MasterIps")
+	}
+
+	k8sClient := config.ClusterFramework.GetClientSets().GetClient()
+	a.isRunning = true
+	a.stopCh = make(chan struct{})
+	a.wg.Add(1)
+	go func() {
+		defer a.wg.Done()
+		for {
+			select {
+			case <-a.stopCh:
+				return
+			case <-time.After(time.Duration(probeDuration)):
+				apiServerAvailable := a.apiServerAvailable(k8sClient, SSHToMasterSupported, config)
+				// Always update the Probes count
+				a.apiAvailabilityMetrics.numProbes++
+				if apiServerAvailable {
+					updateSuccessMetrics(a.apiAvailabilityMetrics)
+				} else {
+					updateFailureMetrics(a.apiAvailabilityMetrics)
+				}
+				return
+			}
+
+		}
+	}()
+	return nil, nil
+}
+
+func updateFailureMetrics(apiAvailabilityMetrics *apiAvailabilityMetrics) {
+	// increment total number of failed probes count, current number of failed probes by 1
+	apiAvailabilityMetrics.numFailures++
+	apiAvailabilityMetrics.consecutiveFailedProbes++
+	// if the previous probe was not a failure, then set the previous probe failure flag to true.
+	// This would be used by next probe
+	if !apiAvailabilityMetrics.prevProbeFailure {
+		apiAvailabilityMetrics.prevProbeFailure = true
+	}
+}
+
+func updateSuccessMetrics(apiAvailabilityMetrics *apiAvailabilityMetrics) {
+	// if the probe succeeded, then increase the number of successful probes count by 1
+	apiAvailabilityMetrics.numSuccess++
+	// if this successful probe was received after failing probe(s)
+	if apiAvailabilityMetrics.prevProbeFailure {
+		// Check if the recently concluded consecutive failed number of probes is
+		// higher than the max consecutive failed number of probes so far
+		// if yes, then Update max consecutive failed probes
+		if apiAvailabilityMetrics.consecutiveFailedProbes > apiAvailabilityMetrics.maxConsecutiveFailedProbes {
+			apiAvailabilityMetrics.maxConsecutiveFailedProbes = apiAvailabilityMetrics.consecutiveFailedProbes
+		}
+		// this probe won't be a failure for the next probe
+		apiAvailabilityMetrics.prevProbeFailure = false
+		// reset current consecutive failed number of probes
+		apiAvailabilityMetrics.consecutiveFailedProbes = 0
+	}
+}
+
+// Execute starts the api-server healthz probe end point from start action and
+// collect metrics in gather.
+func (a *apiAvailabilityMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+
+	SSHToMasterSupported := config.ClusterFramework.GetClusterConfig().SSHToMasterSupported
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return nil, err
+	}
+
+	probeDuration, err := util.GetInt(config.Params, "duration")
```

Comment:
duration isn't the best name - it's rather frequency maybe?

Reply:
We should also default to something sane, maybe 1s?

Reply:
True. Did both.